### PR TITLE
Update factory data

### PR DESCRIPTION
**Pull request by jacobmcnamee**
Update `factory_data_tool` and others to support new factory data parameters. Replace `serial_number` with `uuid`.

TODO:
- [x] Update U-Boot https://github.com/swift-nav/u-boot-xlnx/pull/12
- [x] Test

/cc @swift-nav/firmware @axlan

### CODE REVIEW

File: package/factory_data_tool/src/factory_data_tool.c [hunk 1]
```diff
@@ -139,15 +170,31 @@ int main(int argc, char *argv[])
...
   if (factory_data_timestamp_get(factory_data, &timestamp) == 0) {
     factory_file_write_u32("timestamp", timestamp);
   }
 
+  uint8_t nap_key[16];
+  if (factory_data_nap_key_get(factory_data, nap_key) == 0) {
+    factory_file_write_hex_string("nap_key", nap_key, sizeof(nap_key));
+  }
+
+  uint8_t mac_address[16];
```
Comment (jacobmcnamee) [inline package/factory_data_tool/src/factory_data_tool.c]: 6, not 16

**Comment by axlan** [comment]
Just tested this with swift-nav/piksi_firmware_private#317 and swift-nav/u-boot-xlnx#12 . Boots and navigates without issue.

**Comment by axlan** [comment]
I updated FACTORY_DATA_SIGNATURE to 0x8e54ae38

**Comment by jacobmcnamee** [comment]
Ok, the new file should be pulled in automatically for the buildroot packages once I update the U-Boot pointer.

File: board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot [hunk 2]
```diff
@@ -23,8 +23,8 @@ copy_from_sd() {
 copy_from_net() {
   local server_ip=$1
   local filename=$2
-  local serial_number=`cat /factory/serial_number`
-  local net_file="/PK$serial_number/$filename"
+  local uuid=`cat /factory/uuid`
+  local net_file="/PK$uuid/$filename"
```
Comment (mfine) [inline board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot]: @jacobmcnamee this naming is a little funky:

`/PKa44a00e2-647a-420c-a7ed-f5975a202e48`

Do we need to preserve the `PK` prefix? Alternatives:

`/PK/a44a00e2-647a-420c-a7ed-f5975a202e48`
`/SN-a44a00e2-647a-420c-a7ed-f5975a202e48`
`/swift-nav/a44a00e2-647a-420c-a7ed-f5975a202e48`
`/swiftnav/a44a00e2-647a-420c-a7ed-f5975a202e48`

I don't have much of a horse in this race.

/cc @fnoble

Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S75dev_boot]: I do not care. Whatever you guys think.

FYI the UUID is stored (and used above) without any dashes or anything.

**Comment by axlan** [comment]
lgtm

**Pull request merged**